### ISSUE TITLE
Different results from hasTextFile and hasJsonFile

### ISSUE THREAD

**Issue by RafalFilipek** [commonalityco/commonality#329]
Hey,

I've created a simple rule:

```
import { hasJsonFile } from "commonality-checks-recommended";

export default hasJsonFile("tsconfig.json");
```

It fails in all my packages.

For example:

```
❯ @opl/shared-feature-ab-test-reporter (10)
✘ fail File "tsconfig.json" does not exist
│      packages/shared/feature/ab-test-reporter/tsconfig.json
```

But if I replace `hasJsonFile` with `hasTextFile` it works like a charm.

Info:

`node` - `20.11.0`
`yarn` - `1.22.19`
`@commitlint/cli` - `18.2.0`
`@commitlint/types` - `18.1.0`
`commonality` - `0.2.56`

Please let me know if I can help anyhow.

Thanks,

Rafał

**Update**

Fun fact. It works for `package.json` file for some reason. :)

**Comment by alec-chernicki** [comment]
@RafalFilipek Thanks for filing the issue! 

I tried to reproduce the issue but was unsuccessful. In this [Stackblitz project](https://stackblitz.com/edit/has-json-tsconfig?file=.commonality%2Fhas-tsconfig.ts), running `commonality check` when the `has-tsconfig` check is applied to all packages yields the expected results. 

<img width="510" alt="Screenshot 2024-02-22 at 9 42 00 PM" src="https://github.com/commonalityco/commonality/assets/6596787/aa9744d6-7dbd-47b2-b9ad-398244d61df3">

<img width="372" alt="Screenshot 2024-02-22 at 9 41 54 PM" src="https://github.com/commonalityco/commonality/assets/6596787/c70ec108-fbd2-4838-8204-4926d62ee7a5">

In one package's directory the `tsconfig.json` file exists and the check passes, in the other the check fails because the file does not exist. 

Would you be able to provide a reproduction either as a fork or a link to a sandbox?

**Comment by RafalFilipek** [comment]
Ok, I managed to solve it.

Below, I am inserting an example content of the `tsconfig.json` file

```JSON
{
  "extends": "@opl/shared-typescript-config/v2-lib.json",
}
```


The problem was that `prettier` treats these files as `jsonc` and adds `,` to the ends of the lines.

There is a dedicated issue for this https://github.com/prettier/prettier/issues/15942

This issue has been resolved in the latest version `3.2.5`

It seems to me that in the commonality project itself, a --debug flag would be useful, which would show all errors caught by `try / catch`.

This particular error was caused by the line return JSON.parse(data);

Thanks,

Rafał
